signal/src/lib.rs

```diff
@@ -681,36 +679,37 @@ pub mod pallet {
 			}
 		}
 
-		fn apply_proposal_actions(proposal: &Proposal<T>, proposal_state: ProposalState) -> ProposalState {
+		fn apply_proposal_actions(proposal: &Proposal<T>, proposal_state: ProposalState) -> Result<ProposalState, DispatchError> {
 			match proposal.proposal_type {
 				ProposalType::Withdrawal => {
-					let campaign_id = proposal.campaign_id.unwrap();
-					let amount = proposal.amount.unwrap();
-					T::Currency::unreserve(
-						// SBP-M2 review: `unwrap`
-						proposal.currency_id.unwrap(),
-						&T::Control::org_treasury_account(&proposal.org_id).unwrap(), amount);
+					let campaign_id = proposal.campaign_id.ok_or(Error::<T>::MissingParameter)?;
```

Comment:
The idea of using the "MissingParameter" error was to show an invalid input from the user calling dispatchable.
Here and in the other examples where I see this error was added, it's not about the user input validation but the code bug. This is partially a reason why there was an "unwrap" before, which is probably not the best way, I agree.

But I suggest using another error for this purpose. Otherwise, we are mixing messages. And it would be hard to debug. Also, I think the error log should be there since it's not expected behavior but a bug.

Reply:
@vayesy wdyt? How we should process errors, which are code bugs and not the expected behavior?

Reply:
Fixed.